compiler/rustc_ast/src/mut_visit.rs

```diff
@@ -36,11 +41,144 @@ mod sealed {
 
 use sealed::MutVisitorResult;
 
+pub(crate) trait MutVisitable<V: MutVisitor> {
+    type Extra: Copy;
+    fn visit_mut(&mut self, visitor: &mut V, extra: Self::Extra);
+}
```

Comment:
Question: hm, part of the motivation is to make it more difficult to accidentally miss visiting stuff, right?

Reply:
Yes. And we actually had holes in visits for spans.